### DISCUSSION TITLE
Doc: asyncio client: code requires Python 3.7+

### DISCUSSION THREAD

**Discussion by bialix** [miguelgrinberg/python-socketio#917] [Q&A] [answered]
**Describe the bug**

https://python-socketio.readthedocs.io/en/latest/intro.html#client-examples
I've tried to run your asyncio client example with Python 3.6 and got traceback:

Traceback (most recent call last):
  File "example-client.py", line 40, in <module>
    asyncio.run(main())
AttributeError: module 'asyncio' has no attribute 'run'

According to Python docs asyncio.run is introduced in 3.7 version.

I have no clear suggestion how to improve your docs maybe mention the code is 3.7+ only.

**Comment by bialix** [comment]
Oh my.

I've changed the code to use verbose:

```
    loop = asyncio.get_event_loop()
    try:
        loop.run_until_complete(main())
    finally:
        loop.close()
```

But now I've got another traceback:

```
> python example-client.py
aiohttp not installed -- cannot make HTTP requests!
Traceback (most recent call last):
  File "example-client.py", line 42, in <module>
    loop.run_until_complete(main())
  File "C:\Python\3.6-64\lib\asyncio\base_events.py", line 484, in run_until_complete
    return future.result()
  File "example-client.", line 35, in main
    await sio.connect("http://127.0.0.1:10069")
  File "...\venv\lib\site-packages\socketio\asyncio_client.py", line 164, in connect
    'One or more namespaces failed to connect')
socketio.exceptions.ConnectionError: One or more namespaces failed to connect
```

**Comment by miguelgrinberg** [comment, marked as answer]
@balix are you reading the installation docs? You have a clear error message that points to the problem:

    aiohttp not installed -- cannot make HTTP requests!

You have not installed the client properly, there's missing stuff. See https://python-socketio.readthedocs.io/en/latest/client.html#installation.

With regards to the first problem, you are correct, the example uses `asyncio.run()` which is 3.7+. You don't have to use that though, as you noticed you can also go with the older start up method, which should work in 3.6. I do not keep track of 3.6 support anymore, so it is possible that something eventually will break, but I believe at this point the library should work fine in 3.6. But I recommend you upgrade to something newer, since I won't be maintaining 3.6 support.
